src/EnergyPlus/UnitarySystem.cc

```diff
@@ -0,0 +1,4296 @@
+          heatRecoveryOutletNodeNum(0), noLoadAirFlowRateRatio(0.0), singleMode(0), multiOrVarSpeedHeatCoil(false), multiOrVarSpeedCoolCoil(false),
+          coolingPartLoadFrac(0.0), heatingPartLoadFrac(0.0), suppHeatPartLoadFrac(0.0), heatCompPartLoadRatio(0.0), speedRatio(0.0), cycRatio(0.0)
+    {
+    }
+
+    void UnitarySys::simulate(std::string const &unitarySystemName,
+                              bool const FirstHVACIteration,
+                              int const &AirLoopNum,
+                              int &CompIndex,
+                              bool &HeatActive,
+                              bool &CoolActive,
+                              int const OAUnitNum,
+                              Real64 const OAUCoilOutTemp,
+                              bool const ZoneEquipment)
+    {
+        int CompOn = 0;
+        int unitarySysNum(-1);
+
+        // Obtains and Allocates unitary system related parameters from input file
+        if (UnitarySystems::getInputFlag) {
+            // Get the unitary system input
+            getUnitarySystemInput();
+            UnitarySystems::getInputFlag = false;
+        }
+
+        // Find the correct unitary system Number
+        if (CompIndex == 0) {
+            // unitarySysNum = UtilityRoutines::FindItemInList(unitarySystemName, UnitarySystems::UnitarySys);
+            unitarySysNum = getUnitarySystemIndex(unitarySystemName);
+            if (unitarySysNum == -1) { // zero based index
+                ShowFatalError("SimUnitarySystem: Unitary System not found=" + unitarySystemName);
+            }
+            CompIndex = unitarySysNum;
+        } else {
+            unitarySysNum = CompIndex;
+            // zero based index
+            if (unitarySysNum > (numUnitarySystems-1) || unitarySysNum < 0) {
+                ShowFatalError("SimUnitarySystem:  Invalid CompIndex passed=" + General::TrimSigDigits(unitarySysNum) + ", Number of Unit Systems=" +
+                               General::TrimSigDigits(numUnitarySystems) + ", Unitary System name=" + unitarySystemName);
+            }
+            // if (CheckEquipName(unitarySysNum)) {
+            //    if (UnitarySystemName != UnitarySystem(UnitarySysNum).Name) {
+            //        ShowFatalError("SimUnitarySystem: Invalid CompIndex passed=" + TrimSigDigits(UnitarySysNum) + ", Unitary System name=" +
+            //                       UnitarySystemName + ", stored Unit Name for that index=" + UnitarySystem(UnitarySysNum).Name);
+            //    }
+            //    CheckEquipName(UnitarySysNum) = false;
+            //}
+        }
+
+        if (!this->myGetInputSuccessfulFlag) {
+            // Need to do inputs again for this and any others
+            UnitarySystems::getInputFlag = true;
+            this->getUnitarySystemInput();
+            UnitarySystems::getInputFlag = false;
+        }
```

Comment:
@EnergyArchmage I have a feeling this has something to do with calling during sizing. Line 177-181 of UnitarySystem.cc on this brance (or 397 - 402 of HVACUnitarySystem.cc). Why would a system not be gotten when called? Or maybe it's zone name, index, etc. that needs to be gotten again? So is the HVAC systems being called during sizing and if yes, then why?

    if (!AirLoopFound && !ZoneEquipmentFound && !OASysFound && !DataHVACGlobals::GetAirPathDataDone) {
       // Unsucessful attempt
        continue;
    } else {
        MyGetInputSuccessfulFlag(UnitarySysNum) = true;
    }

Reply:
@rraustad This is not because of sizing, this is because unitary system can be both zone equipment and airloop equipment.  This was a hack to get it working when I added support for multiple airloops service a zone (I think that's when it became a problem.  What needs to happen here is for the input data processing to be separated from the init-type checks that go looking for matching nodes to figure out if this is zone equipment or airloop equipment and then it does all sorts of other checks.

Reply:
Got it. It may be necessary to call a getInput multiple times to get all the data.

Reply:
I think it would be better to call getinput only once, and fill all of the input data, but separate out the various checks that go looking at node connections into a separate init function which only tries to initialize the currently called piece of equipment.  That way it knows if CurZoneEquip or AirloopNum is active and it should have the information it needs at that point.